mantidproject/mantid
# ISIS Reflectometry: Reprocess rows with changed data

This PR adds several new features to the ISIS Reflectometry interface:
- While reduction is paused, selecting new rows of data will cause them to be processed after resuming while deselected ones and already processed data items will be ignored. This is as opposed to previously where the interface simply reprocessed all selected rows, regardless of whether they had been processed before.
- Manually altering data in a previously processed row while paused will cause that row and its containing group to be set unprocessed.
- Adding/removing rows from a group while paused will set that group unprocessed.

**To test:**
- Code review.
- For manual testing:
  - Open the interface from Interfaces->Reflectometry->ISIS Reflectometry
  - Change instrument to `POLREF` and enter investigation id `1520249` (you may need to login to the ICAT catalog). Transfer all runs to the table.
  - Select several groups and process them. Pause midway through and select some new additional groups before processing again. Fully processed groups that were selected should be ignored and the new groups should be processed too.
  - Select several groups, process them and as before, pause midway through. In a previous fully-processed group, change the data of some of the rows. When you resume again, these rows (as well as their containing groups) should be set unprocessed.
- Repeating same steps as before, add/remove some rows from groups while paused. The affected groups should be set unprocessed,



Fixes #19970.

**Release Notes** 

*Release notes have been updated.*

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
